src/main/java/seedu/address/logic/parser/DeleteCategoryCommandParser.java

```diff
@@ -0,0 +1,49 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.Messages.MESSAGE_CANNOT_DELETE_NAME;
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_CATEGORY;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.DeleteCategoryCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parser class for deletecategorycommand
+ */
+public class DeleteCategoryCommandParser implements Parser<DeleteCategoryCommand> {
+    /**
+     * Creates the deletecategorycommand
+     *
+     * @param args our arguments
+     * @return the command
+     * @throws ParseException if no index
+     */
+    @Override
```

Comment:
Try to give some description of what your code does for us to understand easily.